supervisor/shared/web_workflow/static/directory.js

```diff
@@ -92,14 +112,45 @@ async function refresh_list() {
         delete_button.disabled = !editable;
         delete_button.onclick = del;
 
-        edit_url = new URL(edit_url, url_base);
-        let edit_link = clone.querySelector(".edit_link");
-        edit_link.href = edit_url
+        if (editable && !f.directory) {
+            edit_url = new URL(edit_url, url_base);
+            let edit_link = clone.querySelector(".edit_link");
+            edit_link.href = edit_url
+        }
+
+        var dataCell = td[2];
+
+        var sortdata = {};
+        sortdata.value = dataCell.textContent.toLowerCase().trim();
+        sortdata.index = index;
+        if (!f.directory) {
+            dataCells.push(sortdata);
+            index += 1;
+        } else {
+            dirCells.push(sortdata);
+            index += 1;
+        }
 
         new_children.push(clone);
     }
+
+    dirCells.sort(compareValues);
```

Comment:
Can you do `data.sort(compareValues)` before the `for (const f of data)` loop? `compareValues` would then also take both name and dir status into account.

That way you are sorting json objects rather than table rows.

Reply:
Good heavens! this turns out to be an embarrassingly simple change. Thanks for all your guidance!